core/arch/arm/plat-stm32mp1/drivers/stm32mp1_clk.c

```diff
@@ -942,6 +981,185 @@ unsigned long stm32_clock_get_rate(unsigned long id)
 	return rate;
 }
 
+/*
+ * Get the parent ID of the target parent clock, for tagging as secure
+ * shared clock dependencies.
+ */
+static int get_parent_id_parent(unsigned int parent_id)
+{
+	enum stm32mp1_parent_sel s = _UNKNOWN_SEL;
+	enum stm32mp1_pll_id pll_id = _PLL_NB;
+	uint32_t p_sel = 0;
+
+	switch (parent_id) {
+	case _ACLK:
+	case _PCLK4:
+	case _PCLK5:
+		s = _ASS_SEL;
+		break;
+	case _PLL1_P:
+	case _PLL1_Q:
+	case _PLL1_R:
+		pll_id = _PLL1;
+		break;
+	case _PLL2_P:
+	case _PLL2_Q:
+	case _PLL2_R:
+		pll_id = _PLL2;
+		break;
+	case _PLL3_P:
+	case _PLL3_Q:
+	case _PLL3_R:
+		pll_id = _PLL3;
+		break;
+	case _PLL4_P:
+	case _PLL4_Q:
+	case _PLL4_R:
+		pll_id = _PLL4;
+		break;
+	case _PCLK1:
+	case _PCLK2:
+	case _HCLK2:
+	case _HCLK6:
+	case _CK_PER:
+	case _CK_MPU:
+	case _CK_MCU:
+	case _USB_PHY_48:
+		/* We do not expected to access these */
+		panic();
+		break;
+	default:
+		/* Other parents have no parent */
+		return -1;
+	}
+
+	if (s != _UNKNOWN_SEL) {
+		const struct stm32mp1_clk_sel *sel = clk_sel_ref(s);
+		uintptr_t rcc_base = stm32_rcc_base();
+
+		p_sel = (io_read32(rcc_base + sel->offset) >> sel->src) &
+			sel->msk;
+
+		if (p_sel < sel->nb_parent)
+			return (int)sel->parent[p_sel];
```

Comment:
Cast not needed.

Reply:
This is already done automatically by the compiler. I can't find anything in the C standard suggesting that it's more safe to do an explicit conversion (cast) instead of relying on automatic conversions in situations like this. If this result cannot fit into the signed type the result is still implementation defined.

Reply:
Apologies. I get confused with misra rules from tf-a... Will remove these casts.